base/regex.jl

```diff
@@ -363,20 +363,36 @@ function endswith(s::SubString{String}, r::Regex)
     return PCRE.exec_r(r.regex, s, 0, r.match_options | PCRE.ENDANCHORED)
 end
 
+
+# https://stackoverflow.com/questions/5663987/how-to-properly-escape-characters-in-regexp
+_is_regex_escape(c) = in(c, raw"()[{*+.$^\|?")
+
 function chopprefix(s::AbstractString, prefix::Regex)
+    pattern = prefix.pattern
+
+    # fast path
+    startswith(s, pattern) && any(!_is_regex_escape, pattern) && return chopprefix(s, pattern)
+
     m = match(prefix, s, firstindex(s), PCRE.ANCHORED)
     m === nothing && return SubString(s)
     return SubString(s, ncodeunits(m.match) + 1)
 end
 
 function chopsuffix(s::AbstractString, suffix::Regex)
+    pattern = suffix.pattern
+
+    # fast path for regexes meant for file endings
+    startswith(pattern, raw"\.") && any(!_is_regex_escape, @view pattern[3:end]) && endswith(s, @view pattern[2:end]) && return chopsuffix(s, @view pattern[2:end])
+
+    # more general fast path
+    endswith(s, pattern) && any(!_is_regex_escape, pattern) && return chopsuffix(s, pattern)
```

Comment:
These "fast paths" are still of dubious quality (combinations of startswith/endswith with a recursive `chopsuffix` call). They also slow down the general case.


As stated before, I'd merge the docstring improvements, but not these code changes, at least not with some thorough benchmarking data showing how (much) they improve cases were they apply, and conversely how they affect cases were they don't apply.